pkg/controller/che/create.go

```diff
@@ -249,6 +168,14 @@ func (r *ReconcileChe) GenerateAndSaveFields(deployContext *deploy.DeployContext
 		}
 	}
 
+	serverExposureStrategy := util.GetValue(deployContext.CheCluster.Spec.Server.ServerExposureStrategy, deploy.DefaultServerExposureStrategy)
+	if len(deployContext.CheCluster.Spec.Server.ServerExposureStrategy) < 1 {
+		deployContext.CheCluster.Spec.Server.ServerExposureStrategy = serverExposureStrategy
```

Comment:
Do we really need to update `spec.server.serverExposureStrategy` with a default value ?
Could we get rid of it ?

Reply:
The same argument could be made for the majority of other fields for which we update the default value. I was just trying to follow suite. Do you see a strong argument not to?

Reply:
Well, we do have several places where sync values back to CR (our legacy).
But later @davidfestal and me realized that it is a bad practice and we try to avoid it as much as possible.

Reply:
I removed this. Thanks for the explanation.